packages/sync-actions/src/types-actions.js

```diff
@@ -15,3 +24,55 @@ export function actionsMapBase(diff, oldObj, newObj, config = {}) {
     shouldOmitEmptyString: config.shouldOmitEmptyString,
   })
 }
+
+export function actionsMapFieldDefinitions(
+  fieldDefinitionsDiff,
+  previous,
+  next,
+  diffPaths
+) {
+  const actions = []
+  forEach(fieldDefinitionsDiff, (diffValue, diffKey) => {
+    const extractedPairs = extractMatchingPairs(
+      diffPaths,
+      diffKey,
+      previous,
+      next
+    )
+
+    if (getIsChangedOperation(diffKey)) {
+      if (Array.isArray(diffValue)) {
+        const deltaValue = diffPatcher.getDeltaValue(diffValue)
+        if (deltaValue.name) {
+          actions.push({
+            action: 'addFieldDefinition',
+            fieldDefinition: deltaValue,
+          })
+        }
+      } else if (diffValue.label) {
+        actions.push({
+          action: 'changeLabel',
+          label: extractedPairs.newObj.label,
+          fieldName: extractedPairs.oldObj.name,
+        })
+      }
+    } else if (getIsRemovedOperation(diffKey)) {
+      if (Array.isArray(diffValue)) {
+        if (diffValue.length === 3 && diffValue[2] === 3) {
```

Comment:
nothing actionable, but I just want to rant that these magic numbers are driving me nuts